### ISSUE TITLE
backup strategy 

### ISSUE THREAD

**Issue by eagles051387** [LMMS/lmms.io#35]
Seeing as we are migrating everything to our new hosting provider. I would like to start the discussion about backup strategy.

My idea is that since we are on a VPS, I have a script that would compress particular directories and put them in a tmp location if i remember correctly. What I can add is the ability to send an email to certain individuals alerting them that the backup is ready for download. All archives would be deleted after 7 days or how ever many days we decide. What do you guys think?

**Comment by Umcaruje** [comment]
Is backup really necessary, considering we have a git repository that's independent of our hosting? Backup may be only necessary for the wiki and the forums.

**Comment by tresf** [comment]
Yeah, it is necessary but would only be for the database stuff really (wiki, forum, lsp).  It has passwords hashes so we can't replicate it just anywhere or else people could brute force passwords.

We should automate some backup scripts on a few peoples computers I'd think... 

-Tres

**Comment by tresf** [comment]
... or we can dump and copy to a service like google drive on lmms.service account?  Not sure how hard that would be.

**Comment by tresf** [comment]
@eagles051387 you've left this hanging for over a year now.  I've emailed NetworkRedux to see what their backup policies are.

> Support,

> We are reevaluating our backup strategy and wanted to know a simple question... 

> If something drastic were to happen to our MySQL databases, does NetworkRedux have a retention plan to retrieve the data in the event of a critical failure or exploit, or should we be rolling our own?

> For the most part, we'd be happy with a week-old snapshot at any given point in time as the data is for hobbyists and none of it is business critical.

> The data which we do not have weekly copies of would be only the MySql data as the code, crontabs and deployment items are all backed-up to our version control system.

If there are any disagreements with this retention plan speak now or forever hold your peace.

**Comment by tresf** [comment]
Per support:

> The R1Soft CDP is our primary tool for backups and we retain last 30 days backups (Full server backup). If any time you need to restore the previous files, we will be able to provide you between the last 30 days backups.

IMO, this is adequate for our needs.  Please feel free to reopen if you disagree.

**Comment by tresf** [comment]
Reopening to document/investigate this with our newer hosting provider.
